### PR TITLE
Implementation of Walkthrough for new users #62

### PR DESCRIPTION
**Pull request by mariaspastef**
@katerinadimatou and I just implemented the Walkthrough for new users with React Joyride. We believe that it makes the first time using Cboard easier and more understandable. We are really happy with the result and we hope you like it too!

### CODE REVIEW

**Review by tomivm — COMMENTED** [review]
Good Work!! I leave some comments with my opinion. 
Wait for the maintainer's review! 
another thing is that you should commit only one -lock file.  package-lock.json or yarn.lock
you can see Shay's comment on this PR https://github.com/cboard-org/cboard/pull/888

File: src/components/Board/Board.component.js
Comment (katerinadimatou) [reply src/components/Board/Board.component.js]: We tried to do this but it did not work for some reason! That is why we tried placing it in every step!

Comment (tomivm) [reply src/components/Board/Board.component.js]: you're right. it no work. I read [here](https://github.com/gilbarbara/react-joyride/issues/271#issuecomment-429366276) that you can set it with styles
```
styles={
            {
              buttonClose: {
                display: 'none',
              },
            }
          }
```

File: src/components/Board/Board.component.js
Comment (tomivm) [reply src/components/Board/Board.component.js:402]: it's a good point! for me, it's okay :) was only an opinion

File: package.json
Comment (sylvansson) [inline package.json:66]: The pipeline failed because Snyk detected a [vulnerability](https://snyk.io/test/npm/react-joyride/2.3.0) with `react-joyride` v2.3.0. I _think_ we can still merge it, but it might be a good idea to ask @shayc to add an exception in Snyk.

File: src/components/Board/Board.component.js
Comment (sylvansson) [inline src/components/Board/Board.component.js]: You should use a boolean here, not a string. It works currently because a non-empty string behaves exactly like `true` (it's "truthy") but if you put `'false'` the button would probably still be hidden.

```javascript
// The button is hidden.
hideCloseButton: 'true'
hideCloseButton: 'false'

// The button is visible.
hideCloseButton: false
```

(Sorry if you already know about truthy and falsy values.)

File: src/components/Board/Board.component.js
Comment (sylvansson) [inline src/components/Board/Board.component.js:412]: This is just a suggestion, but since most of the code is the same whether the board is locked or not, you could define `Joyride` once and use ternary expressions. Something like:

```javascript
<Joyride
  callback={data => {
    const { status } = data;
    if ([STATUS.FINISHED, STATUS.SKIPPED].includes(status)) {
      this.setState(
        isLocked ? { runLiveUnlock: false } : { runLiveHelp: false }
      );
    }
  }}
  steps={isLocked ? firststep : steps}
  run={isLocked ? runLiveUnlock : runLiveHelp}
  ...
/>
```

I'm not a frontend engineer though so let's see what the experts say first :-)

**Review by sylvansson — COMMENTED** [review]
This looks great. I'm excited to try it :-)

File: src/components/Board/Board.component.js
Comment (katerinadimatou) [reply src/components/Board/Board.component.js]: We tried that too, it really does hide the close button but only from the first message again, then it is shown in the next steps!

Comment (katerinadimatou) [reply src/components/Board/Board.component.js]: No we did not know about truthy and falsy values! Thank you so much, it was very insightful! I will update it!

**Comment by katerinadimatou** [comment]
I noticed today that there was a conflict with yarn.lock file but I did not know a way to resolve it safely. Have you encountered this before? @tomivm @sylvansson

File: src/components/Board/Board.component.js
Comment (katerinadimatou) [reply src/components/Board/Board.component.js:412]: I believe you are right. Because of the fact that we are not so familiar yet with JavaScript and React we took it step by step, but if this is better for the code we will change it of course!

**Comment by sylvansson** [comment]
@katerinadimatou @mariaspastef As @tomivm mentioned it's not a good idea to have multiple lock files, but since `master` currently has both `package-lock.json` and `yarn.lock` we should just fix the merge conflict. Yarn actually detects and fixes merge conflicts automatically, so you should be able to do something like this:

```bash
# The remote might not be called `origin` for you.
git pull origin master
yarn install
git add yarn.lock
git commit --no-edit
git push
```

This worked locally for me, but let me know if you have any issues.

**Comment by katerinadimatou** [comment]
@sylvansson I hope I resolved it!

**Comment by sylvansson** [comment]
@katerinadimatou Looks like you did 🙌

File: package.json
Comment (shayc) [reply package.json:66]: @sylvansson I added you to Snyk as a member.. you should have all the permissions. If you want I can take a look at it in a few days and add the exception.

**Comment by shayc** [comment]
Hi @katerinadimatou and @mariaspastef, I'm going through the PR, will add comments soon.
@sylvansson I marked approved Snyk warnings.
@tomivm thanks for helping!

File: src/components/Board/Board.component.js
Comment (shayc) [inline src/components/Board/Board.component.js]: These comments should be deleted, we want to keep the code as lean as we can, most of the properties are self explanatory, also usually when people change code they forget to update the comment.. so can be a maintenance issue in the future.

File: src/components/Board/Board.component.js
Comment (shayc) [inline src/components/Board/Board.component.js]: We want to be [D.R.Y.](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) as much as we can.
The `styles` object is duplicated (being used twice), should be just a variable, maybe:

```js
const joyRideStyles = {
  options: {
    ...
  }
};
```

File: src/components/Board/Board.component.js
Comment (shayc) [inline src/components/Board/Board.component.js]: Missing translated string "End tour", "Close tour"

File: src/components/UI/LiveHelpButton/LiveHelpButton.js
Comment (shayc) [inline src/components/UI/LiveHelpButton/LiveHelpButton.js]: I think we can delete the `id`? Didn't find anyone using it (not CSS or JS)

File: src/translations/src/cboard.json
Comment (shayc) [inline src/translations/src/cboard.json]: "Are you ready for a ride across Cboard app and its awesome features?"

Should be:

"Are you ready for a tour across Cboard app and its awesome features?"

File: src/translations/src/cboard.json
Comment (shayc) [inline src/translations/src/cboard.json]: "Press the lock 4 times to unlock your options and settings!"
Should be:

"Press the lock button four times to unlock your options and settings!"

File: src/translations/src/cboard.json
Comment (shayc) [inline src/translations/src/cboard.json]: "Sign in or sign up to personalise your communicator!"

Should be (simplicity):

"Sign in to personalize your communicator!"

File: src/translations/src/cboard.json
Comment (shayc) [inline src/translations/src/cboard.json:239]: "Live Help ride"

Should be:

"Live help tour"

File: src/translations/src/cboard.json
Comment (shayc) [inline src/translations/src/cboard.json]: In general, apps  (on Windows, Mac, Android, etc.) do not use exclamation marks in their text.
Maybe we can just use a period "." instead?

File: src/components/UI/LiveHelpButton/LiveHelpButton.js
Comment (shayc) [inline src/components/UI/LiveHelpButton/LiveHelpButton.js]: Should be: [`PropTypes.node`](https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes)

File: src/components/Board/Board.component.js
Comment (shayc) [inline src/components/Board/Board.component.js]: Since `runLiveHelp` and `runLiveUnlock` are boolean variables, it is best practice to prefix them with `should`.
This will help with readability.

Example:
`shouldRunLiveHelp: false`

File: src/components/Board/Board.component.js
Comment (shayc) [inline src/components/Board/Board.component.js]: `hideCloseButton` is a boolean, a better name would be `isCloseButtonHidden` ("hideCloseButton" is a confusing name for a boolean since it implies an action).

File: src/components/Board/Board.component.js
Comment (shayc) [inline src/components/Board/Board.component.js]: We want to keep the state as small as we can (performance reasons), and since the joyride steps are static and do not change it is better to keep the object a property on the class and reference it later in the code as `this.steps`.

File: src/components/Board/Board.component.js
Comment (shayc) [inline src/components/Board/Board.component.js]: We want to remove `firststep` from the `state` object and put in as a class property (for performance reasons).

File: src/components/Board/Board.component.js
Comment (shayc) [inline src/components/Board/Board.component.js]: I think this property can be deleted? I don't think we use it..

**Review by shayc — CHANGES_REQUESTED** [review]
Overall I like the feature very much, there are quite a few comments from me, but it was expected as this is your first code PR. Don't let this discourage you, you guys did a wonderful job, most of my comments are cosmetics.

File: src/components/UI/LiveHelpButton/LiveHelpButton.js
Comment (katerinadimatou) [reply src/components/UI/LiveHelpButton/LiveHelpButton.js]: @tomivm what do you think about that?

File: src/components/Board/Board.component.js
Comment (tomivm) [reply src/components/Board/Board.component.js]: `hideCloseButton` is a prop from Joyride Library. it shows or hides the close icon button in the corner of the modal. for defect, it's no hide.

File: src/components/UI/LiveHelpButton/LiveHelpButton.js
Comment (tomivm) [reply src/components/UI/LiveHelpButton/LiveHelpButton.js]: @shayc is correct! what i mean we no need that id.

Comment (katerinadimatou) [reply src/components/UI/LiveHelpButton/LiveHelpButton.js]: Ok! I deleted it!

File: src/components/Board/Board.component.js
Comment (katerinadimatou) [reply src/components/Board/Board.component.js]: You are right! They are deleted now!

File: src/components/Board/Board.component.js
Comment (katerinadimatou) [reply src/components/Board/Board.component.js]: I made styles a variable named joyRideStyles.

File: src/components/Board/Board.component.js
Comment (katerinadimatou) [reply src/components/Board/Board.component.js]: Added to board.messages.js & cboard.json!

File: src/translations/src/cboard.json
Comment (katerinadimatou) [reply src/translations/src/cboard.json]: I modified the message!

File: src/translations/src/cboard.json
Comment (katerinadimatou) [reply src/translations/src/cboard.json]: I modified the messages and replaced all "!" with ".", except from the welcome message ("Welcome to Cboard!")

File: src/components/UI/LiveHelpButton/LiveHelpButton.js
Comment (katerinadimatou) [reply src/components/UI/LiveHelpButton/LiveHelpButton.js]: Corrected!

File: src/components/Board/Board.component.js
Comment (katerinadimatou) [reply src/components/Board/Board.component.js]: I added the prefix in both variables!

File: src/components/Board/Board.component.js
Comment (katerinadimatou) [reply src/components/Board/Board.component.js]: Yes, I think so too. 
Screenshot 2021-05-28 at 10 16 45 AM
https://docs.react-joyride.com/props

File: src/components/Board/Board.component.js
Comment (katerinadimatou) [reply src/components/Board/Board.component.js]: You are right! I made both steps class properties, and modified their names to unlockedHelpSteps & lockedHelpSteps as we talked about!

File: src/components/Board/Board.component.js
Comment (katerinadimatou) [reply src/components/Board/Board.component.js]: Yes, of course, it was forgotten. Deleted!

File: src/components/Board/Board.component.js
Comment (katerinadimatou) [reply src/components/Board/Board.component.js:402]: We actually took your advice and add the disableOverlayClose = {true} to make it less complicated! So now the user can either press "Close tour" or "Next" until the tour is finished!

File: package.json
Comment (sylvansson) [reply package.json:66]: Thanks @shayc. I actually don't see `react-joyride` in the list of vulnerabilities so it's not immediately obvious what caused the security check to fail. I'll take a closer look this weekend.

**Review by martinbedouret — APPROVED** [review]

**Comment by martinbedouret** [comment]
Thank you @mariaspastef and @katerinadimatou !
Glad to see the reviews from everyone here!

**Pull request merged**
